programs/vote/src/vote_state/mod.rs

```diff
@@ -623,23 +646,36 @@ pub fn process_new_vote_state(
 
     // `new_vote_state` passed all the checks, finalize the change by rewriting
     // our state.
+
+    // New slots not in the current vote must have their vote latency set.  If "timely vote credits" is not enabled,
+    // then 0 is stored for the latency, which will result in 1 credit per slot when credits are calculated at the
+    // time that the slot is rooted.
+    if feature_set
+        .map(|f| f.is_active(&feature_set::timely_vote_credits::id()))
+        .unwrap_or(false)
+    {
+        while new_vote_state_index < new_state.len() {
+            let new_vote = &mut new_state[new_vote_state_index];
```

Comment:
I think the assumption here is that `new_vote_state_index` always points to the earliest new vote in the new vote state? Note there are some subtle cases where this may not be true.

For instance, this is a perfectly legal sequence of vote state updates


1. `{(slot: 5, lockout: 1)}`
2. `{(slot: 4, lockout 2), (slot: 5, lockout: 1)}`

Specifically, the later update introduced a new *earlier* slot than previously. This is legal because the second vote state is still *stricter* than the first one. 

However, using the `new_vote_state_index` will skip the newly introduced slot `4`. More specifically `new_vote_state_index` will point to:

1. the latest new slot `N` in `new_state`
2. where `N` is also greater than the *latest* vote `C` in the current vote state.

And thus will miss new slots less than the latest vote `C`

Reply:
Wow so vote state update allows retroactive voting -- for earlier slots than previously voted on?  That's new.  The regular Vote tx doesn't support that.  OK I'll figure out how to account for this.

Reply:
Also I'll add a test case for that.

Reply:
> Wow so vote state update allows retroactive voting -- for earlier slots than previously voted on

Yeah, this was to account for cases where votes land out of order, which was the primary reason for introducing the new VoteStateUpdate.

> Also I'll add a test case for that.

Awesome, thanks!